### DISCUSSION TITLE
Conditionally hiding relationships in BudiDB

### DISCUSSION THREAD

**Discussion by Bitello** [Budibase/budibase#9463] [Help] [answered]
Hi!

I'm using BudiDB and I have the following difficulty:
- I have a Projects table related to a Managers table, which in turn is related to an Employees table;
- A Manager can be in N Projects;
- A Project must have several Managers, but only one with a null end date attribute (the current Manager);

How, in a Project form table, do I display only the Manager who has a null end date (without displaying the entire history of managers who have a completed end date)?

And, in the case that all Manager records have the end date filled in, present, in a Project form table, as "if there was no Manager" in the relationship (even if they exist in the history, but with the date end filled)?

Grateful

**Comment by melohagan** [comment, marked as answer]
Hey @Bitello 

Could you share some screenshots of what you've tried please? It'll help me to visualise the problem!

**Reply by Bitello** [reply]
Thanks for replying, @melohagan.

My intention is to create a functionality where, by clicking on a line icon in the Inspection (Fiscalização) table, it is possible to edit or include its Coordinator.

To choose whether to open the registration or editing side page, I evaluate whether the relationship with the Coordination (Coordenação) table is filled in or not.
![image](https://user-images.githubusercontent.com/22456825/216108382-29f307be-930b-4013-98ac-dd53011772db.png)

![image](https://user-images.githubusercontent.com/22456825/216108773-7c544590-e7af-496b-a970-db9e9d69a6dc.png)

![image](https://user-images.githubusercontent.com/22456825/216108906-23abb9dc-6798-456d-b448-c0ae2bbb84a8.png)

My problem occurs when I edit the Coordinator assigning an end date. The relationship with the Coordination table obviously continues to exist. Therefore, I cannot open a side page to insert a new Coordinator.

At the beginning of this project, I chose to use BudiDB, to explore the potential of the tool. If I were using MS Sql Server (which is within my comfort zone) I would create a View relating the Inspection and Coordination tables not returning the records that have dtEnd filled in.
With BudiDB (within my understanding limitation) I can generate a View for a table, and not for a "query".

I was thinking of using (but first I need to learn to use) automations. In creating a new table that would be treated as a View with relationships.

There may be an "obvious" way to achieve the result I'm looking for, but I can't see it.

**Reply by melohagan** [reply]
Ah I think I understand the issue. In the internal Budibase DB you cannot filter by relationship. 

So in your case you want to show the 'Edit' side panel if a Coordinator exists AND the Coordinator.end_date is null ?

You could use a SQL external DB to filter on `Coordinator.end_date`. 

Otherwise you may want to split this out across a couple of screens instead (https://docs.budibase.com/docs/url-parameters)

**Comment by helioguardabaxo** [comment]
@melohagan 
Budibase DB cannot filter by formula field too. Right?

**Reply by melohagan** [reply]
You can filter by **static** formulas, which are available in the internal Budibase DB only.

**Reply by helioguardabaxo** [reply]
Thanks! Worked here!

**Comment by Bitello** [comment]
I managed to work around the situation (but I had to stop the tests due to the bug that does not allow editing table functions).

In the "Coordination" BudiDB table I created a function type column that returns 0 (create a new record) or 1 (edit the record).

In the Table component with the "Inspection" data, I inserted a "column" with a datasource that does a lookup with the "Coordination" BudiDB table (filtering the data provider by the relationship between the "Inspection" and "Coordination" tables) and evaluates whether access to the "Coordination" insertion or editing page.

![image](https://user-images.githubusercontent.com/22456825/217942261-c0ba5ff3-2aa0-4340-9ee7-d21149000a36.png)

I forgot, when it comes to insertion, in the "Inspection Table" component, to remove the name of the coordinator (and that's where I'm stuck waiting for the column editing bug to be corrected)
